File: scala/FizzBuzz/src/main/scala/core/Rule.scala
package core

class Rule(denominator: Int, result: String) {

  def giveResult(): String = {
    result
  }

  def appliesTo(number: Int): Boolean = {
    isDivisibleBy(number, denominator)
  }

  private def isDivisibleBy(number: Int, denominator: Int): Boolean = {
    number % denominator == 0
  }
}

File: java/FizzBuzzWithMaven/src/main/java/Rule.java
public class Rule {
    private final int denominator;
    private final String result;

    public Rule(int denominator, String result) {
        this.denominator = denominator;
        this.result = result;
    }

    private static boolean isDivisibleBy(int number, int denominator) {
        return number % denominator == 0;
    }

    public String getResult() {
        return result;
    }

    public boolean appliesTo(int number) {
        return isDivisibleBy(number, denominator);
    }
}
